### DISCUSSION TITLE
Add update button to an outdated diff

### DISCUSSION THREAD

**Discussion by heyngra** [ppy/osu#21884] [Ideas] [answered]
# Problem
Imagine this: You make yourself a practice diff on a map, and you edit the main diff and also you edit a secondary diff made out of that one. You have two local difficulties, which one is derived from an uploaded beatmap, and one is made entirely by you. 

Now, that you're done with training on the first difficulty (the one from an uploaded beatmap), you would want to try your skills on a full version of the map. And here's the problem. You can't update only one difficulty. On osu!stable, you would just go onto the global ranks, and click `Update to latest version`. It would **only** update this one diff, without touching any other. Sadly in osu!lazer, there is only an update button on the entire mapset. And it makes the entire beatmapset redownload, which means that all practice diffs are nuked, and not only the difficulty that we want to try.

# My suggestions to fix the problem

1. Make a smaller update button on each difficulty that is out to date (and available to download from web, so you won't try to update unexisting, custom made diff), that updates only this difficulty and not the entire mapset

2. Make the beatmapset update button not nuke all difficulties, and only redownload the ones that need update

**Comment by bdach** [comment, marked as answer]
This use case would be better supported by something like https://github.com/ppy/osu/discussions/13013.
